### ISSUE TITLE
clippy::if_then_panic warning (nightly)

### ISSUE THREAD

**Issue by Bromeon** [brendanzab/approx#73]
This is not a pressing issue yet, but it appears in a nightly version of clippy, so you might be interested in addressing the lint ahead of time.

I'm using this simple snippet:
```rs
use approx::assert_relative_eq;

fn main() {
    assert_relative_eq!(1.0, 2.0);
}
```

When running `cargo +nightly clippy`, I get:
```rs
warning: only a `panic!` in `if`-then statement
 --> src\main.rs:4:2
  |
4 |     assert_relative_eq!(1.0, 2.0);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::if_then_panic)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_then_panic
  = note: this warning originates in the macro `__assert_approx` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This occurred in the CI of the downstream project [godot-rust](https://github.com/godot-rust/godot-rust), which treats clippy warnings as errors (the nightly workflow still passes though).

---

It has to do with how the `__assert_approx` macro is implemented.
Basically, the lint tries to detect the structure
```rs
if !cond { panic!(msg); }
```
and transform it into
```rs
assert!(cond, msg);
```

This would be one option to fix it, another would be to `#[allow(clippy::if_then_panic)]` the relevant statements.

I'm using `rustc 1.57.0-nightly (5b210643e 2021-10-11)`.

**Comment by EFanZh** [comment]
I think #72 has fixed this issue, so maybe this can be closed?

**Comment by Bromeon** [comment]
The Clippy lint has been renamed and moved categories in the meantime: https://github.com/rust-lang/rust-clippy/pull/7810

However, it still seems to occur with `approx v0.5.0`.
A crate with only this code:
```rs
fn main() {
    approx::assert_relative_eq!(1.0, 1.0);
}
```
results in:
```rs
cargo clippy -- -D clippy::pedantic
    Checking ctest v0.1.0 (<path/to/project>)
error: only a `panic!` in `if`-then statement
 --> src\main.rs:2:5
  |
2 |     approx::assert_relative_eq!(1.0, 1.0);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D clippy::manual-assert` implied by `-D clippy::pedantic`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_assert
  = note: this error originates in the macro `__assert_approx` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `<crate>` due to previous error
```

**Comment by EFanZh** [comment]
The fix was just merged a few days ago, new version containing the fix hasn’t been released yet.
